index.html

```diff
@@ -86,25 +86,311 @@ <h3>Use-case #4: Avoiding "Hall of Mirrors"</h3>
+      </ul>
+      <div class="note">
+        There is disagreement on whether actions should be specified here or in a separate document.
+      </div>
+    </section>
+    <section id="captured-side-actions">
+      <h2>Captured Side for Actions</h2>
+      <p>
+        Applications in top-level documents can declare the [=capture actions=]
+        they support, if any. They would typically do so before even knowing if
+        they are being captured. These actions enable the user to control the
+        progression of the captured session without leaving the capturer document.
+        Supported actions are declared by calling {{MediaDevices/setSupportedCaptureActions}}
+        with an array of the names of actions the application is prepared to respond to.
+      </p>
+      <section id="set-supported-capture-actions">
+        <h3>Registering and responding to capture actions</h3>
+        <p>
+          {{MediaDevices}} is extended with a method - {{MediaDevices/setSupportedCaptureActions}} -
+          which accepts an array of {{DOMString}}s. By calling this method, an application
+          registers with the user agent a set of zero or more [=capture actions=] it wishes to
+          respond to.
+        </p>
+        <p>
+          <dfn>Capture actions</dfn> are values defined in {{CaptureAction}}.
+          They are meant to be interpreted as instructions from the user (through the user agent)
+          to control the advancement of the presentation of the captured session, however the
+          captured application wishes to define this. Actions are designed to originate from the
+          user through interactive controls provided by the capturer document, where they
+          [=consume user activation=].
+        </p>
+        <pre class="idl">
+          partial interface MediaDevices {
+            undefined setSupportedCaptureActions(sequence&lt;DOMString&gt; actions);
+            attribute EventHandler oncaptureaction;
+          };
+
+          enum CaptureAction {
+            "next",
+            "previous",
+            "first",
+            "last"
+          };
+        </pre>
+        <dl data-link-for="MediaDevices" data-dfn-for="MediaDevices" class="methods">
+          <dt>
+            <dfn>setSupportedCaptureActions</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST run the following steps:</p>
+            <ol>
+              <li>
+                If the [=relevant settings object=]'s
+                [=environment settings object/responsible document=] is either not
+                [=Document/fully active=] or its [=browsing context=] is not a
+                [=top-level browsing context=], then throw {{InvalidAccessError}}.
+              </li>
+              <li>
+                Let |actions| be the method's first argument.
+              </li>
+              <li>
+                If |actions| is non-empty, and this method was previously
```

Comment:
It means "ever" on [this](https://webidl.spec.whatwg.org/#this) `MediaDevices` object (which is local to the document). I.e. (only) the third code line you show would fail.

The intent is trying to forbid changing the set of actions.

As to why: allowing it creates a novel cross-origin communication channel again #11, so conservatively it seemed simpler to forbid it, at least initially, to reduce the number of things people might object to.

I'm not closed to there being some utility of changing actions like you describe (e.g. not showing a "previous" button on the first slide), but this begs other questions like needing an event to learn of updates, so I figured we'd start small.

Reply:
(Please also see other comment thread on this topic.)
Cross-document navigation of the captured tab would always allow the set of declared actions to change.
This means that the capturer would anyway need an event for new-actions-declated.
I think we can't start smaller than that. I'd remove this limitation.